XiEditor/StyleMap.swift

```diff
@@ -14,13 +14,87 @@
 
 import Cocoa
 
+extension NSFont {
+    /// If the font is monospace, returns the width of a character, else returns 0.
+    func characterWidth() -> CGFloat {
+        if self.isFixedPitch {
+            let characters = [UniChar(0x20)]
+            var glyphs = [CGGlyph(0)]
+            if CTFontGetGlyphsForCharacters(self, characters, &glyphs, 1) {
+                let advance = CTFontGetAdvancesForGlyphs(self, .horizontal, glyphs, nil, 1)
+                return CGFloat(advance)
+            }
+        }
+        return 0
+    }
+}
+
 /// A represents a given text style.
 struct Style {
-    var fgColor: NSColor;
-    var bgColor: NSColor;
-    var weight: UInt16;
-    var underline: Bool;
-    var italic: Bool;
+    var font: NSFont
+    var fgColor: NSColor
+    var bgColor: NSColor
+    var setTextBgColor: Bool
```

Comment:
`wantsTextBgColor`? 

But I'm not sure I totally understand: text background color should be a distinct property from global background color. The concept of a global background color should exist outside of `Style`, which should more accurately be thought of as `TextStyle` (This renaming might be helpful?). How we end up doing non-text UI styling is something to discuss.